web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1049,6 +1049,239 @@ $.extend(fixmystreet.set_up, {
+            $reports.each(function() {
+                var id = parseInt( $(this).data('reportId'), 10 );
+                var alert_url = '/alert/subscribe?id=' + encodeURIComponent(id);
+                var $button = $('<a>').addClass('btn btn--small btn--primary btn--get-updates');
+                $button.text(translation_strings.get_updates);
+                $button.attr('href', alert_url);
+                $(this).find('.item-list__item--expandable__actions').append($button);
+            });
+        }
+    }
+
+    function render_duplicate_pins(api_response) {
+        var markers = fixmystreet.maps.markers_list( api_response.pins, true );
+        fixmystreet.markers.removeFeatures( current_duplicate_markers );
+        fixmystreet.markers.addFeatures( markers );
+        current_duplicate_markers = markers;
+    }
+
+    function remove_duplicate_list(cb) {
+        var animations = [];
+
+        animations.push( $.Deferred() );
+        $('#js-duplicate-reports').slideUp(function(){
+            $(this).addClass('hidden');
+            $(this).find('ul').empty();
+            animations[0].resolve();
+        });
+        if ( $('#problem_form').length ) {
+            animations.push( $.Deferred() );
+            $('.js-hide-if-invalid-category').slideDown(function(){
+                animations[1].resolve();
+            });
+        }
+
+        $.when.apply(this, animations).then(cb);
+    }
+
+    function remove_duplicate_pins() {
+        fixmystreet.markers.removeFeatures( current_duplicate_markers );
+    }
+
+    function inspect_form_state_change() {
+        // The duplicate report list only makes sense when state is 'duplicate'
+        if ($(this).val() !== "duplicate") {
+            $("#js-duplicate-reports").addClass("hidden");
+            return;
+        } else {
+            $("#js-duplicate-reports").removeClass("hidden");
+        }
+        // If this report is already marked as a duplicate of another, then
+        // there's no need to refresh the list of duplicate reports
+        var duplicate_of = $("#report_inspect_form [name=duplicate_of]").val();
+        if (!!duplicate_of) {
+            return;
+        }
+        refresh_duplicate_list();
+    }
+
+    var category_changing = false;
+    function problem_form_category_change() {
+        // Annoyingly this event seems to fire a few times in quick succession,
+        // so set a flag to avoid multiple overlapping refreshes.
+        if (category_changing) { return; }
+        category_changing = true;
+
+        refresh_duplicate_list().always(function(){
+            // Wait an extra second until we allow another reload.
+            setTimeout(function(){
+                category_changing = false;
+            }, 1000);
+        });
+    }
+
+    // Want to show potential duplicates when a regular user starts a new report.
+    // TODO: This also fires after you move the pin while making a report. Do we want that?
```

Comment:
I guess we do, but perhaps if they haven't started filling in the textboxes/uploaded an image?